scripts/pre-commit.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# Find CUDA and CPP files and apply clang-format 
+find ./ -iname *.h -or -iname *.cuh -or -iname *.cu | xargs clang-format -i -style=file
```

Comment:
what about .c / .cpp files?

Reply:
> what about .c / .cpp files?

Can definitely add. I didn't see that we have any files that are strictly .c/.cpp right now so I didn't put them in.

Reply:
@vhnatyk I kept the single script for this since each language tries to write their own scripts to the .git/hooks directory and they were conflicting with each other. I added a README section on how to configure the hook scripts